niet/__init__.py

```diff
@@ -179,4 +184,7 @@ def main():
     if args.in_place and infilename:
         out_file = infilename
 
+    if to_delete:
+        result = actions.delete(result, to_delete)
```

Comment:
Before you need to check if we get a dict:
```python
if isinstance(result, dict):
```

Reply:
Yep, you're right, but I'd rather make the test in the function itself, it is more relevant considering the recursive part ...